osu.Game.Rulesets.Osu/Mods/OsuModSpunOut.cs

```diff
@@ -2,21 +2,86 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
 using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.StateChanges;
 using osu.Game.Graphics;
 using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
+using osu.Game.Rulesets.Osu.Replays;
+using osu.Game.Rulesets.UI;
+using static osu.Game.Input.Handlers.ReplayInputHandler;
 
 namespace osu.Game.Rulesets.Osu.Mods
 {
-    public class OsuModSpunOut : Mod
+    public class OsuModSpunOut : Mod, IUpdatableByPlayfield, IApplicableToDrawableRuleset<OsuHitObject>
     {
         public override string Name => "Spun Out";
         public override string Acronym => "SO";
-        public override IconUsage? Icon => OsuIcon.ModSpunout;
-        public override ModType Type => ModType.DifficultyReduction;
-        public override string Description => @"Spinners will be automatically completed.";
+        public override IconUsage? Icon => OsuIcon.ModSpunOut;
+        public override ModType Type => ModType.DifficultyIncrease;
+        public override string Description => @"Spinners will be automatically completed";
+        public override Type[] IncompatibleMods => base.IncompatibleMods.Append(typeof(OsuModAutopilot)).ToArray();
         public override double ScoreMultiplier => 0.9;
-        public override bool Ranked => true;
-        public override Type[] IncompatibleMods => new[] { typeof(ModAutoplay), typeof(OsuModAutopilot) };
+        private List<OsuReplayFrame> replayFrames;
+        private int currentFrame;
+        public void Update(Playfield playfield)
+        {
+            bool requiresHold = false;
+            inputManager.AllowUserCursorMovement = true;
+            foreach (var drawable in playfield.HitObjectContainer.AliveObjects)
+            {
+                inputManager.AllowUserCursorMovement = true;
```

Comment:
These repeated assignments of the same values to the same variable are not necessary.